File: frontend/lib/models/document.dart
class DocumentModel {
  final int id;
  final String filename;
  final String? mimeType;
  final String originalPath;
  final String? processedPath;
  final String? ocrText;
  final String status;
  final DateTime createdAt;

  DocumentModel({
    required this.id,
    required this.filename,
    this.mimeType,
    required this.originalPath,
    this.processedPath,
    this.ocrText,
    required this.status,
    required this.createdAt,
  });

  factory DocumentModel.fromJson(Map<String, dynamic> json) {
    return DocumentModel(
      id: json['id'],
      filename: json['filename'],
      mimeType: json['mime_type'],
      originalPath: json['original_path'],
      processedPath: json['processed_path'],
      ocrText: json['ocr_text'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: backend/app/models/document.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database.session import Base

class Document(Base):
    __tablename__ = "documents"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"))
    filename = Column(String, index=True)
    mime_type = Column(String)
    original_path = Column(String)
    base_processed_path = Column(String, nullable=True)  # Perspective-corrected base image
    processed_path = Column(String, nullable=True)  # Current image (may have filters applied)
    parent_document_id = Column(Integer, ForeignKey("documents.id"), nullable=True)  # For cropped images
    ocr_text = Column(Text, nullable=True)
    document_type = Column(String, default="typed")  # typed, handwritten, other
    status = Column(String, default="uploaded") # uploaded, processing, completed, failed
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    owner = relationship("User", back_populates="documents")

# Update User model to include relationship
